### PR TITLE
When terminating detach the connection from request/response objects

### PR DESCRIPTION
**Pull request by violetagg**
Related to #3416, #3367

### CODE REVIEW

**Comment by violetagg** [comment]
@chemicL wait with the review - some of the tests are failing ...

File: reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java [hunk 1]
```diff
@@ -681,4 +693,111 @@ static OnSetup empty() {
...
+				promise.setFailure(new UnsupportedOperationException());
+			}
+		}
+	}
+
+	static final class DisposedConnection implements Connection {
+
+		final Channel channel;
+
+		DisposedConnection(Channel copy) {
```
Comment (chemicL) [inline reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java:794]: I struggle with the name a bit. It's not really a copy as the name would imply. The behaviour you expect is to copy so you're treating a name as a verb, which is confusing. Am I getting it wrong? Perhaps `Channel actual` would be indicative that we're working with something that wraps the actual `Channel`. Then the same comment would apply to the argument of `DisposedChannel`'s constructor.

File: reactor-netty-http-brave/src/main/java/reactor/netty/http/brave/TracingHttpServerDecorator.java [hunk 2]
```diff
@@ -266,27 +266,25 @@ static final class TracingMapHandle implements BiFunction<Mono<Void>, Connection
...
-			                               else {
-			                                   eventLoop.execute(() -> handler.handleSend(braveResponse, localSpan));
-			                               }
-			                       });
+			                       if (eventLoop.inEventLoop()) {
+			                           handler.handleSend(braveResponse, localSpan);
+			                       }
+			                       else {
+			                           eventLoop.execute(() -> handler.handleSend(braveResponse, localSpan));
+			                       }
```
Comment (chemicL) [inline reactor-netty-http-brave/src/main/java/reactor/netty/http/brave/TracingHttpServerDecorator.java:287]: Took me a while to understand what's going on here, but I understand that at this point the `response.withConnection `lambda can potentially have the `DisposedConnection` which means there's no event loop associated any more so you are essentially capturing it in the closure through the above reference, correct?

File: reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java [hunk 3]
```diff
@@ -3401,4 +3403,46 @@ static void testIssue3285SendRequest(HttpClient client, @Nullable Class<? extend
...
+		                            boolean result = connWeakRef.get().get() == null;
+		                            if (!result) {
+		                                System.gc();
+		                            }
+		                            return result;
+		                        })
+		                        .switchIfEmpty(Mono.error(new RuntimeException("failed"))
+		                        .flatMap(l -> Mono.empty())))
+		                .then()
+		                .contextWrite(Context.of(res.getClass(), res)))
```
Comment (chemicL) [inline reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java:3443]: I would love to see a comment that this serves the purpose of addressing another concern. While the main test flow validates that the connection is freed once response is retrieved as `res` reference, the `contextWrite` store ensures also that the res in `Context` also doesn't preserve the `Connection` reference. Am I understanding it right?

**Review by chemicL — APPROVED** [review]
Looks good, I just left some questions and requests for comments :)

File: reactor-netty-http-brave/src/main/java/reactor/netty/http/brave/TracingHttpServerDecorator.java (hunk 2 above)
Comment (violetagg) [reply reactor-netty-http-brave/src/main/java/reactor/netty/http/brave/TracingHttpServerDecorator.java:287]: correct

File: reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java (hunk 3 above)
Comment (violetagg) [reply reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java:3443]: correct

File: reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java (hunk 1 above)
Comment (violetagg) [reply reactor-netty-core/src/main/java/reactor/netty/channel/ChannelOperations.java]: change applied

**Review by chemicL — APPROVED** [review]
Thanks for the update! 🚢

**Comment by violetagg** [comment]
@chemicL Thanks for the review!

**Pull request merged**